### PR TITLE
feat: allow kotlin test in api, maps to kotest

### PR DESCRIPTION
**Pull request by sdelamo**
Close: https://github.com/micronaut-projects/micronaut-starter/issues/261

### CODE REVIEW

File: starter-api/src/test/groovy/io/micronaut/starter/api/create/CreateControllerKotlinTestSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import javax.inject.Singleton
+
+class CreateControllerKotlinTestSpec extends EmbeddedServerSpecification {
+
+    @Override
+    Map<String, Object> getConfiguration() {
+        super.configuration + [
+                'spec.name': 'CreateControllerKotlinTestSpec'
+        ]
+    }
```
Comment (ilopmar) [inline starter-api/src/test/groovy/io/micronaut/starter/api/create/CreateControllerKotlinTestSpec.groovy:30]: You can override `getSpecName()`

File: starter-api/src/test/groovy/io/micronaut/starter/api/preview/PreviewControllerKotlinTestSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+import javax.inject.Singleton
+
+class PreviewControllerKotlinTestSpec extends EmbeddedServerSpecification {
+    @Override
+    Map<String, Object> getConfiguration() {
+        super.configuration + [
+                'spec.name': 'PreviewControllerKotlinTestSpec'
+        ]
+    }
```
Comment (ilopmar) [inline starter-api/src/test/groovy/io/micronaut/starter/api/preview/PreviewControllerKotlinTestSpec.groovy:27]: You can override `getSpecName()`

**Review by ilopmar — CHANGES_REQUESTED** [review]

File: starter-api/src/test/groovy/io/micronaut/starter/api/create/CreateControllerKotlinTestSpec.groovy (hunk 1 above)
Comment (sdelamo) [reply starter-api/src/test/groovy/io/micronaut/starter/api/create/CreateControllerKotlinTestSpec.groovy:30]: Yup. I missed that.

File: starter-api/src/test/groovy/io/micronaut/starter/api/preview/PreviewControllerKotlinTestSpec.groovy (hunk 2 above)
Comment (sdelamo) [reply starter-api/src/test/groovy/io/micronaut/starter/api/preview/PreviewControllerKotlinTestSpec.groovy:27]: Yup, I i missed that.

**Review by ilopmar — APPROVED** [review]

**Pull request merged**